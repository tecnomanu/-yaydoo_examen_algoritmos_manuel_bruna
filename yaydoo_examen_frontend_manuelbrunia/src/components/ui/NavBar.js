import React, { useState } from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
    const handleNavCollapse = () => setIsNavCollapsed(!isNavCollapsed);
    const [isNavCollapsed, setIsNavCollapsed] = useState(true);

    return (
        <nav className="navbar navbar-expand-md navbar-dark bg-dark">
            <div className="container-fluid">
                <Link
                    className="navbar-brand"
                    to="/"
                >
                    Algoritmos
                </Link>
                <button className="navbar-toggler"
                    type="button"
                    data-bs-toggle="collapse"
                    data-bs-target="#navbarNavDropdown"
                    aria-controls="navbarNavDropdown"
                    aria-expanded={!isNavCollapsed ? true : false}
                    aria-label="Toggle navigation"
                    onClick={handleNavCollapse}>
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className={`${isNavCollapsed ? 'collapse' : ''} navbar-collapse`} id="navbarNavDropdown">
                    <ul className="navbar-nav">

                        <li className="nav-item">
                            <NavLink
                                className={({ isActive }) => 'nav-link ' + (isActive ? 'active' : '')}
                                to="/users"
                            >
                                Usuarios
                            </NavLink>

                        </li>
                        {/* <li className="nav-item">
                            <NavLink
                                className={({ isActive }) => 'nav-item nav-link ' + (isActive ? 'active' : '')}
                                to="/users/create"
                            >
                                Crear un usuario nuevo
                            </NavLink>
                        </li> */}
                    </ul>
                </div>
            </div>
        </nav>
    )
}