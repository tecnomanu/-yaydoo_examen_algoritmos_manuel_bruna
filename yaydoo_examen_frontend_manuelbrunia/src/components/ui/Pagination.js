import React, { useEffect, useState } from 'react'
import ReactDOM from 'react-dom';
import { useNavigate } from 'react-router';
import ReactPaginate from 'react-paginate';

export const Pagination = ({data}) => {
    const navigate = useNavigate();
    const [pageCount, setPageCount] = useState(0);
    const [itemOffset, setItemOffset] = useState(0);
    const itemsPerPage = 10;

    useEffect(() => {
        const endOffset = itemOffset + itemsPerPage;
        setPageCount(Math.ceil((data?.total || 0) / itemsPerPage));
    }, [data]);

    const handlePageClick = (event) => {
        const newOffset = (event.selected * itemsPerPage) % data.total;
        setItemOffset(newOffset);
        navigate(`?page=${event.selected + 1}`);
    };

    return (
        <nav className="navigation mt-2">
            <ReactPaginate
                className="pagination justify-content-center"
                pageClassName="page-item"
                pageLinkClassName="page-link"
                previousClassName="page-item"
                nextClassName="page-item"
                previousLinkClassName="page-link"
                nextLinkClassName="page-link"
                activeClassName="active"
                breakLabel="..."
                nextLabel="Sig. >"
                onPageChange={handlePageClick}
                pageRangeDisplayed={5}
                pageCount={pageCount}
                previousLabel="< Ant."
                renderOnZeroPageCount={null}
            />
        </nav>
    )
}
