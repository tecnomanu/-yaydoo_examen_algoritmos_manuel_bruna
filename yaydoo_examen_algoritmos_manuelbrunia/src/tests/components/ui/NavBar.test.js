import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { Navbar } from '../../../components/ui/NavBar';

describe("Test on <Navbar/>", ()=>{
    test('SnapShot of <Navbar/>', ()=>{
        const wrappper = shallow(<Navbar />);

        expect(wrappper).toMatchSnapshot();
    })
})