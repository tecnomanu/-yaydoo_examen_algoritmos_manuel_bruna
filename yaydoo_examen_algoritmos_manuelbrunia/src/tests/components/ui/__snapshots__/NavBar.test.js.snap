// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Navbar/> SnapShot of <Navbar/> 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <Link
      className="navbar-brand"
      to="/"
    >
      Algoritmos
    </Link>
    <button
      aria-controls="navbarNavDropdown"
      aria-expanded={false}
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarNavDropdown"
      data-bs-toggle="collapse"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNavDropdown"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <NavLink
            className={[Function]}
            to="/test-1"
          >
            Alice vs Bob
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            className={[Function]}
            to="/test-2"
          >
            Cuenta Palabras
          </NavLink>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
